### PR TITLE
Feature/pe lifecycle events progress tracker

### PR DESCRIPTION
**Pull request by 13pake**

### CODE REVIEW

File: src/pe-lifecycle-events-progress-tracker/config.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  ],
+  "contributors": [
+    "Trevor Roberts"
+  ],
+  "keywords": [
+    "Widget"
+  ],
+  "name": "pe-lifecycle-events-progress-tracker",
+  "publishing-goal": "Update set",
+  "solution-details": "",
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/config.json]: - [ ] fill out and complete solution details

File: src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        '</div>' +
+        '</div>';
+
+  angular.module('donutDirective', []).directive('donutDirective', function($timeout) {
+    return {
+      restrict: 'AE',
+      // templateUrl: function (element, attrs) {
+      //     return attrs.href;
+      // },
+      template: donutTemplate,
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js]: - [ ] remove unused commented out code

File: src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+          colors = {
+            fill: scope.colorFill,
+            bg: scope.colorBg
+          };
+        }
+        scope.randomID = randomId;
+        /*  $timeout(function(){
+                 calculateDonut(scope.width, percent, scope.showValue,
+                 scope.showValueAsCount, scope.count, colors, scope.inverted, randomId);
+                 }, 1);*/
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js]: - [ ] remove commented out unused code

File: src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        showValueAsCount: '=',
+        width: '=',
+        colorFill: '@',
+        colorBg: '@',
+        inverted: '=',
+        fontSize: '@',
+        fontWeight: '='
+      },
+      link: function(scope, element, attrs) {
+        //console.log("displayvalue " + scope.displayvalue);
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/ui-script/pe-lifecycle-event-progress-tracker.module.u-client-script.js]: - [ ] remove commented out console.log

File: src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-css.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
+.panel-heading {
+    display: block;
+  }
+  
+  .dropdown-menu {
+    border-radius: 0px; 
+
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-css.scss]: - [ ] remove empty line

File: src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-body-html-template.html [hunk 6]
```diff
@@ -1,3 +1,44 @@
...
+
+  <div class="panel-body">
+    <div class="flex-container">
+      <div class="donut text-center" ng-repeat="donut in data.donuts | filter:{event: c.selectedEvent}">
+        <a href="{{donut.href}}">
+        <donut-directive data-count="donut.complete" data-displayvalue="{{donut.percentage}}" data-total="donut.total"
+                         data-show-value="true" data-show-value-as-count="true" data-width="data.chart_width" data-font-size="{{data.font_size}}"
+                         data-font-weight="data.font_weight" data-color-fill="{{data.primary_color}}" data-color-bg="{{data.background_color}}">
+        </donut-directive>
+          <h4>{{donut.title}}</h4>
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-body-html-template.html]: Consider one way binding if the context allows it. 
https://github.com/platform-experience/serviceportal-best-practice#one-way-binding

File: src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-server-script.js [hunk 7]
```diff
@@ -1,4 +1,137 @@
...
+      grRCA.addQuery('target', grTable.getUniqueValue());
+      grRCA.query();
+      if (grRCA.next()) {
+        data.errorLink = '/nav_to.do?uri=/sys_restricted_caller_access.do?sys_id=' + grRCA.getUniqueValue();
+      } else {
+        data.errorLink = '/nav_to.do?uri=/sys_restricted_caller_access_list.do?sysparm_query=source=' + widgetID;
+      }
+      break;
+    }
+  }
```
Comment (asc2683) [inline src/pe-lifecycle-events-progress-tracker/widget/pe-lifecycle-events-progress-tracker.u-server-script.js]: A lot going on in this while loop. Consider refactoring.

**Comment by stevengregory** [comment]
hey @13pake, did you see the change requests from @asc2683?

**Comment by stevengregory** [comment]
No feedback for months on this—closing out the PR.

**Pull request closed without merging**